src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php

```diff
@@ -409,4 +411,131 @@ public function supportsDescendantType(string $type): bool
     {
         return $this->getClassName() === $type;
     }
+
+    private function createChildCollectionsQueryBuilder(string $alias, int $rootCollectionId): QueryBuilder
+    {
+        $rootCollectionAlias = $alias . '_rootCollection';
+
+        return $this->createQueryBuilder($alias)
+            ->innerJoin(
+                CollectionInterface::class,
+                $rootCollectionAlias,
+                Join::WITH,
+                $alias . '.lft > ' . $rootCollectionAlias . '.lft AND ' . $alias . '.rgt < ' . $rootCollectionAlias . '.rgt'
+            )
+            ->where($rootCollectionAlias . '.id = :id')
+            ->setParameter('id', $rootCollectionId);
+    }
+
+    /**
+     * @return array<array{id: int, resourceKey: string, depth: int}>
+     */
+    public function findChildCollectionResourcesOfRootCollection(int $rootCollectionId): array
+    {
+        return $this->createChildCollectionsQueryBuilder('collection', $rootCollectionId)
+            ->select('collection.id AS id')
+            ->addSelect('\'' . CollectionInterface::RESOURCE_KEY . '\' AS resourceKey')
+            ->addSelect('collection.depth AS depth')
+            ->distinct()
+            ->orderBy('collection.id', 'ASC')
+            ->getQuery()
+            ->getArrayResult();
+    }
+
+    /**
+     * @return int[]
+     */
+    public function findChildCollectionIdsOfRootCollection(int $rootCollectionId): array
```

Comment:
think this method is not needed the `mediaRepository` should just get a `rootCollectionId`